File: backend/app/schemas/mtd.py
from datetime import datetime
from typing import Dict, List, Optional
from pydantic import BaseModel


class PathRotationInfo(BaseModel):
    dynamic_path: str
    target_handler: str
    created_at: datetime
    status: str  # "active" or "deprecated"


class MTDStatusResponse(BaseModel):
    mtd_enabled: bool
    current_seed: str
    active_routes: Dict[str, str]  # dynamic_path -> real_path mapping
    decoy_paths: List[str]
    rotation_interval_seconds: int
    last_rotation: Optional[datetime] = None
    next_rotation_in_seconds: Optional[float] = None
    rotation_history: List[PathRotationInfo] = []


class HoneypotLogEntry(BaseModel):
    id: Optional[int] = None
    decoy_path_triggered: str
    ip_address: str
    user_agent: Optional[str] = None
    timestamp: datetime
    headers_logged: Dict[str, str]

File: frontend/src/api/mtd.ts
import api from "./client";

export interface PathRotationInfo {
  dynamic_path: string;
  target_handler: string;
  created_at: string;
  status: string;
}

export interface MTDStatusResponse {
  mtd_enabled: boolean;
  current_seed: string;
  active_routes: Record<string, string>;
  decoy_paths: string[];
  rotation_interval_seconds: number;
  last_rotation: string | null;
  next_rotation_in_seconds: number | null;
  rotation_history: PathRotationInfo[];
}

export const getMTDStatus = async (): Promise<MTDStatusResponse> => {
  const response = await api.get("/api/v1/mtd/status");
  return response.data;
};

export const triggerMTDRotation = async (): Promise<{ message: string; status: MTDStatusResponse }> => {
  const response = await api.post("/api/v1/mtd/rotate");
  return response.data;
};
